File: algorithms/55. Jump Game/solution.cc
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int final = nums.size() - 1;
        for ( int i = 0, j = 0; i <= j; ++i ) {
            j = max(j, nums[i] + i);
            if ( j >= final )
                return true;
        }
        return false;
    }
};

File: algorithms/55. Jump Game/solution.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        i = final = 0
        while i <= final:
            final = max(final, nums[i] + i)
            if final >= len(nums) - 1:
                return True
            i += 1
        return False

# 75 / 75 test cases passed.
# Status: Accepted
# Runtime: 62 ms
# beats 48.32 %
